lib/irb/source_finder.rb

```diff
@@ -77,6 +77,8 @@ def method_target(owner_receiver, s_count, method, type)
         target_method = target_method.super_method if target_method
       end
       target_method.nil? ? nil : target_method.source_location
+    rescue NameError
```

Comment:
We can do more sophisticated rescue for individual `instance_method` and `method` calls. But for now with all the tests I think we can assume it's unlikely the logic itself would cause `NoMethodError`.